samples/account-linking/csharp/Source/Controllers/AccountLinkingController.cs

```diff
@@ -0,0 +1,71 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Teams.Samples.AccountLinking.OAuth;
+using Microsoft.Identity.Web.Resource;
+using System.Security.Claims;
+using Microsoft.Identity.Web;
+using System.Net;
+
+namespace Microsoft.Teams.Samples.AccountLinking.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public sealed class AccountLinkingController : ControllerBase
+{
+    private readonly OAuthTokenProvider _tokenProvider;
+
+    private readonly ILogger<AccountLinkingController> _logger;
+
+    public AccountLinkingController(
+        ILogger<AccountLinkingController> logger,
+        OAuthTokenProvider tokenProvider)
+    {
+        _logger = logger;
+        _tokenProvider = tokenProvider;
+    }
+
+    [Authorize]
+    [RequiredScope(RequiredScopesConfigurationKey = "AzureAd:Scopes")]
+    [HttpPut("link")]
+    public async Task<IActionResult> LinkAccountsAsync(
+        [FromBody] AccountLinkRequestBody tokenClaimRequest)
+    {
+        var userId = User.FindFirstValue(ClaimConstants.ObjectId);
```

Comment:
ObjectId and TenantId are the "old" object and tenant id, looks like Oid and Tid are the new ones. I'm not sure if there is a timeline or deprecation path, but perhaps it makes sense to do something like `Oid ?? ObjectId`, just to reduce the chances od someone hitting an issue where their token contains the new Oid/Tid (I actually think Bhavya hit this while testing)

https://docs.microsoft.com/en-us/dotnet/api/microsoft.identity.web.claimconstants?view=azure-dotnet

Reply:
This is tricky because the Microsoft.Identity.Web library is populating Object / Tenant by default and I'd rather not get into the world of covering [Claims mapping](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/claims?view=aspnetcore-6.0) as part fo this. 

I added a note in the README to configure the app for v2 tokens. I'll take this as an item for the next iteration/v2